lua/weapons/laserpointer/shared.lua

```diff
@@ -19,3 +19,32 @@ SWEP.Secondary.ClipSize = -1
 SWEP.Secondary.DefaultClip = -1
 SWEP.Secondary.Automatic = false
 SWEP.Secondary.Ammo = "none"
+
+function SWEP:UpdateHeldStatus()
+	local owner = IsValid(self:GetOwner()) and self:GetOwner()
+	local isValidOwner = owner ~= false
+
+	self.IsHeld = isValidOwner
+	self.Wielder = isValidOwner and owner or nil
+end
+
+function SWEP:OwnerChanged()
+	timer.Simple(0, function() self:UpdateHeldStatus() end)
+end
```

Comment:
I'd just use and check `self:GetOwner()` everywhere its needed. No need to store it. Could also lead to issues if OwnerChanged fails to trigger for whatever reason.

Reply:
I agree with you. The motivation here was that we'll have to do many "Is valid owner" checks, so I wanted to make the code simpler with a simple `if self.IsHeld`